### PR TITLE
fix(sdk-coin-ada): rejecting ada outputs below 1 ADA

### PR DESCRIPTION
**Pull request by dgm003**
Ticket: CSHLD-459

**Description:**
ADA transactions with output amounts less than 1ADA used to get stuck in the sendqueue.

**Fix:**
Added validation in the ADA transaction builder to enforce the Cardano protocol minimum of 1 ADA per output:

- Receiver outputs below 1 ADA are rejected at build time
- Dust change (< 1 ADA) is absorbed into the fee instead of creating an unspendable output
- ADA-only consolidations that would produce a sub-1-ADA output are rejected early, allowing the sweep to skip that address and continue

**Testing**
All (164) unit tests passing. New tests cover the boundary conditions (999,999 lovelace rejected, 1,000,000 lovelace allowed, dust change absorbed, sub-1-ADA consolidation rejected).

### CODE REVIEW

File: modules/sdk-coin-ada/src/lib/transactionBuilder.ts [hunk 1]
```diff
@@ -297,8 +297,13 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
 
           change = change.checked_sub(minAmountNeededForAssetOutput);
         } else {
-          // Native coin send
+          // Native coin send — reject if below Cardano's minimum output (BabbageOutputTooSmallUTxO)
           const amount = CardanoWasm.BigNum.from_str(receiverAmount);
+          if (amount.less_than(CardanoWasm.BigNum.from_str(MIN_ADA_TRANSFER))) {
```
Comment (abhishekagrawal080) [inline modules/sdk-coin-ada/src/lib/transactionBuilder.ts:302]: do we need this exception for if change.less_than(CardanoWasm.BigNum.from_str(MIN_ADA_TRANSFER))) as well?

**Comment by dgm003** [comment]
Closing this PR as ive made the necessary changes in BGMS repo.
